### DISCUSSION TITLE
v2.5.0

### DISCUSSION THREAD

**Discussion by the1812** [the1812/Bilibili-Evolved#3750] [分享 / Share]
✨新增
- 新增功能 `隐藏动态评论预览`. (#3322)
> 隐藏动态评论按钮上方的精选评论预览.

- 新增功能 `显示视频投稿时间`. (PR #3727 by [wisokey](https://github.com/wisokey))
> 为视频播放页面的推荐列表中的视频添加显示视频投稿时间.

- 新增功能 `展开选集列表`. (#3380)
> 总是完全展开视频选集列表, 注意对番剧无效.

🐛修复
- 由于 Chrome / Firefox / Safari 均已支持自适应滚动条, `夜间模式` 不再覆盖默认滚动条样式, 设置的组件详情不再隐藏滚动条. (#3370)
- 修复 `下载字幕` 得到的 ASS 文件格式不正确. (#3688)
- 修复 `播放前显示封面` 在视频暂停时仍然显示封面. (#3698)
- `自定义顶栏` 的弹窗改为有焦点时不自动收起, 以防止在其中打字时意外收起. (#3703)
- 修复 `动态过滤器` 的关键词过滤在首页和顶栏中有时无法生效的问题, 以及对转发动态无效的问题.
- 修复 b 站消息会被 `自定义顶栏` 遮挡的问题. (#3702)
- 修复 `动态反折叠` 有时失效的问题.
- 修复 `极简首页` 中合作投稿样式的问题.
- 尝试修复 `document.body is null` 报错导致脚本无法加载的问题. (#3728)
- 修复 `下载弹幕` 无法读取 3.x 播放器的弹幕设置, 以及文件中的弹幕乱序的问题. (#3739)
- 修复 `自定义顶栏` 在搜索页没有正确隐藏原版顶栏. (#3731)

☕开发者相关
- 包管理器更换为 pnpm, **请删除 node_modules 并重新安装依赖 (`pnpm install`)**.
- ESLint 加入了 Prettier 规则. (#3729)
- 更多的组件更换为 define API 声明. (PR #3682, PR #3697 by [timongh](https://github.com/timongh))
- 改善了动态的 `addMenuItem` API 稳定性.
- 添加了 `createPostHook` API, 可以让原函数先执行再让钩子函数执行.

<hr /><em>This discussion was created from the release <a href='https://github.com/the1812/Bilibili-Evolved/releases/tag/v2.5.0'>v2.5.0</a>.</em>

**Comment by Geeyun-JY3** [comment]
功能的命名有些混乱且和 B 站不统一啊，感觉用户可能会混淆：#3752

目前 v2 的这个 [`展开选集标题`](https://github.com/the1812/Bilibili-Evolved/blob/31ada0aeeeabfae2dc614f32b0eada78b9a8cfd4/doc/features/features.md#展开选集标题) 功能和 v1 时一样，除了展开视频分 P 标题（即会换行），实际上还会像 v2.5.0 新加的这个 [`展开选集列表`](https://github.com/the1812/Bilibili-Evolved/blob/31ada0aeeeabfae2dc614f32b0eada78b9a8cfd4/doc/features/features.md#展开选集列表)  展开合集列表一样展开视频分 P 列表；
不过目前是 `展开选集列表` 只能展开合集列表，`展开选集标题` 只能展开分 P 列表。

B 站目前把分 P 列表叫「选集列表」，按现在这个功能命名法用户就可能以为 `展开选集列表` 功能可以展开分 P 列表。

建议把展开分 P 列表和合集列表做成一个功能（分 P 和合集列表分别给个开关？我个人无所谓），并让 `展开选集标题`（或者别用「展开」这个字眼？主要是怕用户混淆）就只是让标题换行，同时支持分 P 和合集列表（即 #692 的第一点）。

另外（v2）最新文档里 [`展开选集列表`](https://github.com/the1812/Bilibili-Evolved/blob/31ada0aeeeabfae2dc614f32b0eada78b9a8cfd4/doc/features/features.md#展开选集列表) 的内部名称还没改过来，还是 `fullEpisodeTitle`（看来这俩确实容易搞混）。https://github.com/the1812/Bilibili-Evolved/commit/a308b46367ddec2ca18eaab2a048400b75595812

***

个人碎碎念：
才发现 v1 的 [`展开选集列表`](https://github.com/the1812/Bilibili-Evolved/blob/9f452209fed54a6e7a9bdf8cb3c3f7c87bb71e5e/doc/features.md#工具:~:text=%E9%9A%90%E8%97%8F%E6%90%9C%E7%B4%A2%E6%8E%A8%E8%8D%90-,%E5%B1%95%E5%BC%80%E9%80%89%E9%9B%86%E5%88%97%E8%A1%A8,-%E7%9B%B4%E6%92%AD%E9%97%B4%E5%8B%8B%E7%AB%A0) 在 v2 改名为 [`展开选集标题`](https://github.com/the1812/Bilibili-Evolved/blob/9f452209fed54a6e7a9bdf8cb3c3f7c87bb71e5e/doc/features/features.md#展开选集标题) 了。

**Reply by the1812** [reply]
命名是有问题，周末弄个v2.5.1改一改

**Comment by Geeyun-JY3** [comment]
目前 [`显示视频投稿时间`](https://github.com/the1812/Bilibili-Evolved/blob/master/doc/features/features.md#显示视频投稿时间) 在[功能](https://github.com/the1812/Bilibili-Evolved/blob/master/doc/features/features.md)列表里的两个 `Stable` 安装链接打开都是显示 `404: Not Found`，直接通过设置面板在线安装 `master` 分支的该组件（脚本本体非 `preview` 版）也会报错，强行安装 `preview` 分支的倒是可以用。

另外[功能](https://github.com/the1812/Bilibili-Evolved/blob/master/doc/features/features.md)列表里所有组件和插件相同分支的 **AltCdn** 和 **GitHub** 的安装链接目前都是一样的。
